pkg/scheduler/algorithm_factory.go

```diff
@@ -389,28 +390,47 @@ func RegisterCustomPriorityFunction(policy schedulerapi.PriorityPolicy, args *pl
 	// generate the priority function, if a custom priority is requested
 	if policy.Argument != nil {
 		if policy.Argument.ServiceAntiAffinity != nil {
+			// We use the ServiceAffinity plugin name for all ServiceAffinity custom priorities.
+			// It may get called multiple times but we essentially only register one instance of
+			// ServiceAffinity priority.
+			// This name is then used to find the registered plugin and run the plugin instead of the priority.
+			name = serviceaffinity.Name
+
+			if configProducerArgs.ServiceAffinityArgs == nil {
+				configProducerArgs.ServiceAffinityArgs = &serviceaffinity.Args{}
+			}
+			configProducerArgs.ServiceAffinityArgs.Labels = append(configProducerArgs.ServiceAffinityArgs.Labels, policy.Argument.ServiceAntiAffinity.Label)
+
+			weight := policy.Weight
+			schedulerFactoryMutex.RLock()
+			if existing, ok := priorityFunctionMap[name]; ok {
+				// If there are n ServiceAffinity priorities in the policy, the weight for the correspoinding
```

Comment:
still not fixed :) corresponding

Reply:
sorry was looking at the correct spell below..